pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -641,7 +641,11 @@ func (c *Reconciler) runNextSchedulableTask(ctx context.Context, pr *v1beta1.Pip
 					return fmt.Errorf("error creating Run called %s for PipelineTask %s from PipelineRun %s: %w", rprt.RunName, rprt.PipelineTask.Name, pr.Name, err)
 				}
 			} else {
-				rprt.TaskRun, err = c.createTaskRun(ctx, rprt, pr, as.StorageBasePath(pr))
+				if rprt.IsFinalTask(pipelineRunFacts) {
```

Comment:
> We need to ensure that a PipelineRun fails if enable-api-fields: stable is set after the pr is submitted but before it executes. This could happen, for example, if the reconciler is busy and processing pipelineruns slowly and an operator flips the enable-api-fields flag between those two events

Hmm... Do we check just before the first reconcile, or at each reconcile cycle? My opinion here is that once the pipelinerun has started, we shouldn't take into account any operator update for that pipelinerun anymore... hence I would add a check of the `APIFields` flag only at the first reconcile. What do you think?

Reply:
> My opinion here is that once the pipelinerun has started, we shouldn't take into account any operator update for that pipelinerun anymore... hence I would add a check of the APIFields flag only at the first reconcile. What do you think?

That's a really great point, I hadn't thought about this breaking pipelines that are mid-run.  I agree with you - checking only alpha usage on the first reconcile makes sense to me.

Where I see this further breaking down is if an operator sets the `stable` flag midway through a pipeline run, and one of the later tasks in a pipeline relies on an alpha feature.  then that task would fail the pipeline.

Also, pinging @vdemeester  @afrittoli  @bobcatfish for any insight from the authors of TEP-0033 here as well.